apps/language_server/lib/language_server/providers/document_symbols.ex

```diff
@@ -54,22 +70,24 @@ defmodule ElixirLS.LanguageServer.Providers.DocumentSymbols do
 
   # Identify and extract the module symbol, and the symbols contained within the module
   defp extract_modules({:__block__, [], ast}) do
-    ast |> Enum.map(&extract_modules(&1)) |> List.flatten()
+    ast |> Enum.map(&extract_modules/1) |> List.flatten()
   end
 
   defp extract_modules({defname, _, _child_ast} = ast)
        when defname in [:defmodule, :defprotocol, :defimpl] do
-    [extract_symbol("", ast)]
+    extract_symbol(ast, %Symbols{module_name: "", symbols: [], custom_defs: []})
```

Comment:
The extra definitions provided by the formatter and the user config would be loaded and passed as the initial value for `custom_defs` here and in the call below, basically